### PR TITLE
Implement docker-compose

### PR DESCRIPTION
**Pull request by asiday**
Changes:
-Change the link for scheduled task in crawler to "country=ch"
-Create docker-compose.yml file with crowler and mariadb
-Create appropriate schema for creating new db
-Add to docker-compose.yml file service.
-Save key as secret in docker-compose.yml

Close#17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,17 @@
-# news-aggregator
-News aggregator
+# News-aggregator
+News aggregator contains of three parts:
++ crawler 
+  + scheduled task with get request from news-api:
+    + https://newsapi.org/v2/top-headlines?country=ch&apiKey=XXX 
+  + requested json files are saved in mariadb
```
Comment (yambay) [inline README.md:6]: Do you really save JSON files in the database? Or maybe from JSON converted entities?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
++ service 
+  + get rows from existing mariadb
```
Comment (yambay) [inline README.md:8]: Get entities. Rows sound too simple

File: README.md [hunk 3]
```diff
@@ -1,2 +1,17 @@
...
+  + return them as json files under the link
+    + localhost:8080/articles
++ mariadb
+
+## Instructions to Run application.
+The application uses docker-compose.yml file to run three images of crawler/service/mariadb as separate docker containers. 
++ Save all sensitive information such as api key in .env file
```
Comment (yambay) [inline README.md:15]: Please, describe which parameters one should put into '''.env''' file.

Comment (yambay) [reply README.md:15]: Maybe an example of an .env file would be good at this point

File: crawler/src/main/java/dev/asida/crawler/schedule/ScheduledTask.java [hunk 4]
```diff
@@ -38,7 +39,8 @@ public ScheduledTask(NewsService service, ArticleRepository repository) {
                 initialDelayString = "${initialDelay.in.milliseconds:5000}")
     public void requestArticles() {
         RequestContext r = new RequestContext();
-        r.getLanguages().add(Language.GERMAN);
+        r.getCountries().add(Country.SWITZERLAND);
+        //r.getLanguages().add(Language.RUSSIAN);
```
Comment (yambay) [inline crawler/src/main/java/dev/asida/crawler/schedule/ScheduledTask.java:43]: Why do you leave commented code here? Please remove

File: crawler/src/main/resources/schema.sql [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
-      url VARCHAR(500),
-      url_to_image VARCHAR(500),
+      description VARCHAR(2000),
+      url TEXT,
+      url_to_image TEXT,
       published_at DATETIME,
-      content VARCHAR(1000),
+      content VARCHAR(2000),
       PRIMARY KEY (id)
 );
```
Comment (yambay) [inline crawler/src/main/resources/schema.sql:11]: Do you need this SQL file here? You moved it into news-aggreagator/scripts, so please remove it from here (or move into test if neccesary)

File: crawler/src/test/java/dev/asida/crawler/schedule/ScheduledTaskTest.java [hunk 6]
```diff
@@ -29,7 +30,8 @@ void init() {
     public void requestScheduled() {
         // given
         RequestContext r = new RequestContext();
-        r.getLanguages().add(Language.GERMAN);
+        r.getCountries().add(Country.SWITZERLAND);
+        //r.getLanguages().add(Language.GERMAN);
```
Comment (yambay) [inline crawler/src/test/java/dev/asida/crawler/schedule/ScheduledTaskTest.java:34]: Remove commented code

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+version: "3.7"
+
+services:
+  crawler:
+    image: "asiday/news-crawler:0.0.3"
+    restart: always
+    environment:
+      KEY: "${API_KEY}"
+      SPRING_DATASOURCE_URL: jdbc:mysql://app_db:3306/news_aggregator
```
Comment (yambay) [inline docker-compose.yml:9]: The database name is already defined as "${MARIADB_DATABASE}", so there is no need to hard code the database name in the data source url. Please use something like 

jdbc:mysql://app_db:3306/${MARIADB_DATABASE}

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    links:
+      - maria_db:app_db
+
+  service:
+    image: "asiday/news-service:0.0.1"
+    ports:
+      - "8080:8080"
+    restart: always
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://app_db:3306/news_aggregator
```
Comment (yambay) [inline docker-compose.yml:25]: Use ${MARIADB_DATABASE} instead of hard coded DB name

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      SPRING_JPA_HIBERNATE_DDL-AUTO: update
+      SPRING_DATASOURCE_INITIALIZATION-MODE: always
+    depends_on:
+      - maria_db
+    links:
+      - maria_db:app_db
+
+  maria_db:
+    image: "mariadb:10.7.3"
+    ports:
```
Comment (yambay) [inline docker-compose.yml:37]: I don't think, that this image should expose any ports. Please, try to remove this ports mapping and test if your applications still work.

File: service/src/main/resources/schema.sql [hunk 10]
```diff
@@ -1,12 +1,11 @@
...
-      id INT NOT NULL AUTO_INCREMENT,
-      author VARCHAR(100),
-      title VARCHAR(500),
-      description VARCHAR(1000),
-      url VARCHAR(500),
-      url_to_image VARCHAR(500),
-      published_at DATETIME,
-      content VARCHAR(1000),
-      PRIMARY KEY (id)
+CREATE TABLE IF NOT EXISTS articles (
```
Comment (yambay) [inline service/src/main/resources/schema.sql:1]: Do you need this SQL file here? You moved it into news-aggreagator/scripts, so please remove it from here (or move into test if neccesary)

**Review by yambay — CHANGES_REQUESTED** [review]

**Comment by yambay** [comment]
By the way. Your commit message starts with "Implement docker-compose". Are you sure, that you implement docker-compose? Or maybe you just make use of docker-compose to start your application?

File: README.md (hunk 1 above)
Comment (asiday) [reply README.md:6]: Of course "JSON converted entities"

File: README.md (hunk 2 above)
Comment (asiday) [reply README.md:8]: Done.

File: README.md (hunk 3 above)
Comment (asiday) [reply README.md:15]: Examples are added.

File: crawler/src/main/java/dev/asida/crawler/schedule/ScheduledTask.java (hunk 4 above)
Comment (asiday) [reply crawler/src/main/java/dev/asida/crawler/schedule/ScheduledTask.java:43]: Removed.

File: crawler/src/test/java/dev/asida/crawler/schedule/ScheduledTaskTest.java (hunk 6 above)
Comment (asiday) [reply crawler/src/test/java/dev/asida/crawler/schedule/ScheduledTaskTest.java:34]: Done

File: docker-compose.yml (hunk 7 above)
Comment (asiday) [reply docker-compose.yml:9]: The URL changed using placeholder.

File: docker-compose.yml (hunk 8 above)
Comment (asiday) [reply docker-compose.yml:25]: Done.

File: docker-compose.yml (hunk 9 above)
Comment (asiday) [reply docker-compose.yml:37]: After removing ports mapping, the app is still working.

File: service/src/main/resources/schema.sql (hunk 10 above)
Comment (asiday) [reply service/src/main/resources/schema.sql:1]: Tried to remove. Unfortunately it is not working.

**Review by yambay — APPROVED** [review]

File: crawler/src/main/resources/schema.sql (hunk 5 above)
Comment (asiday) [reply crawler/src/main/resources/schema.sql]: The schema.sql file is used by Spring Data under the hood.

**Pull request merged**